src/nxdoc/nuxeo-server/platform-services/comments.md

```diff
@@ -34,27 +35,74 @@ They hold:
 [`Comment`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Comment.html) interface has been created to handle comments in Nuxeo. The interface is used as based type in [comment service](#java-api) and also defines how comments are serialized in JSON in order to use them with the [REST API](#json-entity-format).
 
 Annotations are stored as regular document of type `Annotation` which inherit from `Comment`. They hold in addition of comments:
+
 - blob xpath
 
 Like comments, annotations have the [`Annotation`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Annotation.html) interface to use them.
 
 Comments and annotations can have the facet `ExternalEntity`. When they have this facet, it can hold in addition:
+
 - external entity id
 - external entity
 - external entity origin</br>
-With this facet, it is possible to handle external content such as annotations made in [Nuxeo Enhanced Viewer]({{page space='nxdoc' page='nuxeo-enhanced-viewer'}}) for instance.
+With this facet, it is possible to handle external content such as annotations made in [ARender]({{page space='nxdoc' page='nuxeo-arender-connector'}}) for instance.
+
+### Current implementation - TreeCommentManager
+
+#### Storage
+
+Comments and annotations are stored as a tree structure in a folderish document of type `CommentRoot` which is located right under the commented document. Comments are _special children_ which means they are not copied with the commented document during copy operations but they are snapshot when the document is checked in.</br>
+Please note comments are not restored when a version of the commented document is checked out.
+
+#### Permissions
+
+Comments and annotations rely on the the [`ACL`](https://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/core/api/security/ACL.html) security model present in Nuxeo. People who have access to a commented document have read access to its comments and can edit their own comments and annotations.
+
+#### Events
+
+All events share the same event context. It contains:
+
+- the parent document in `parentComment` which may be a comment or the commented document
+- the comment in `comment_document`
+- the comment text in `comment`
+- the category `commentCategory` in `category`
+
+A `commentAdded` event is fired when creating a comment or annotation.
+
+A `commentUpdated` event is fired when updating a comment or annotation.
+
+A `commentRemoved` event is fired when deleting a comment or annotation.
+
+#### Autosubscription
+
+When a first comment is added to a document, the document author is subscribed automatically to the `commentAdded` event on the document.</br>
+When a user adds his first comment on a document. He is subscribed automatically to the `commentAdded` event on the document.</br>
```

Comment:
I thought the paragraph is a bit complex. Something like _author is automatically subscribed to the notifications of its comment_ sounds a bit less complex.

Furthermore, the second part is composed of two sentences which are linked (one can't be understand without the other), can you reword it?

Reply:
I propose this:
>Author is subscribed automatically to the `commentAdded` event on his documents.\
Comment Author is subscribed automatically to the `commentAdded` event on the commented document.\